plugins/modules/files/archive.py

```diff
@@ -131,7 +131,7 @@
 '''
 
 RETURN = r'''
-state:
+dest_state:
```

Comment:
I discussed this in the description: 
>The result dict will never show this information however because implementing path or dest as options causes AnsibleModule to overwrite state to the file state. (See here)
https://github.com/ansible/ansible/blob/e6a7bc854cabc99eff1b46109fcde7eb28aaac23/lib/ansible/module_utils/basic.py#L1221-L1228

I don't consider it a breaking change in this PR because that result value will work exactly as before.

Reply:
Sorry, I realized the `state` is still coming through. It would be nice then to *add* the `dest_state` to the documentation instead of *replacing* `state` with it.

Reply:
Yes, please keep some documentation for `state`, and add `version_added` to `dest_state`.

Also note that this is a change that cannot be backported to stable-2 as a bugfix, since this counts as a new feature.

Reply:
That's a good point. Let me check some ansible builtins which implement the file options and see if there's a standard definition of that state value.

Reply:
In that case @felixfontein I can do the following:
- Leave `dest_state` out of the result dict for this PR (And remove related tests)
- Update the description for `state` to be accurate
- Push a subsequent PR with the `dest_state` value

Does that sound reasonable?

Reply:
@Ajpantuso sounds good to me!